doc/adr/0012-commit-to-datatypes-in-cardano-api.rst

```diff
@@ -0,0 +1,141 @@
+
+This was desirable in order to start designing a way to build Plutus application before the ``cardano-ledger`` actually supported the Alonzo-era features.
+Of course, this resulted in the unintented consequence that we used ``TxInfo`` types (types that are designed to be used in Plutus scripts) in off-chain code.
+This wouldn't has been a problem if there was 1:1 relationship between on-chain and off-chain types.
+However, that presumption is wrong.
+
+Let's take the example of the ``TxOut`` representation of ``plutus-ledger-api`` for ``PlutusV2``.
+
+.. code-block:: haskell
+
+  data TxOut = TxOut {
+    ...
+    txOutReferenceScript :: Maybe ScriptHash
+    }
+
+As we can see, the ``TxOut`` can optionnaly store the ``ScriptHash`` of the referenced script.
+However, that is *not* the adequate representation of a ``TxOut`` in a
+transaction given the ``cardano-ledger`` specification.
+The off-chain ``TxOut`` should instead be:
+
+.. code-block:: haskell
+
+  data TxOut = TxOut {
+    ...
+    txOutReferenceScript :: Maybe Script
+    }
+
+where the reference script field can store that *actual* script, not just the hash.
+
+This proved that we need to start moving away from `plutus-ledger-api` types in
+the off-chain part of Plutus applications, especially in components like the
+emulator and the chain indexer.
+
+Decision
+--------
+
+* We will remove our datatype representation of a Cardano transaction
+  (``Ledger.Tx.Internal.Tx`` in ``plutus-ledger``) and fully commit to
+  ``Cardano.Api.Tx.Tx era`` in the codebase.
+
+* We will replace any use of ``plutus-ledger-api`` types by ``cardano-api`` types
+  whenever we work with the off-chain part of Plutus applications.
+  For instance, the ``plutus-contract`` emulator and types in the
+  ``Plutus.Contract.Request`` module of ``plutus-contract`` will be updated to
+  use ``cardano-api`` types.
+  However, the datatypes in ``Ledger.Constraints.TxConstraints`` will continue
+  to use ``plutus-ledger-api`` types because the constraints are used to
+  generate both Plutus scripts and transactions.
+  Therefore, there should be no breaking change on the API for writing Plutus
+  applications.
+
+* We will create a ``cardano-api-extended`` cabal project which will contain
+  features and utilies on top of the ``cardano-api`` package.
+  A similar idea has emerged with hydra-cardano-api_.
+  This package will contain:
+
+    * type synonyms for working with the latest era
+
+    * a simplified and working transaction balancing function (mainly the
+      ``Ledger.Fee.makeAutoBalancedTransaction`` in ``plutus-apps``)
+
+    * validation rules (most of what's in the current ``Ledger.Validation``)
+
+* We will improve ``cardano-api`` through ``cardano-api-extended`` and regularly
+  push changes upstream when possible.
+
+* We will have packages in ``plutus-apps`` depend on ``cardano-api-extended`` which
+  will re-export the modules from the hydra-cardano-api_ package.
```

Comment:
Not sure of the meaning here, will packages `plutus-apps` re-export the modules, or will `cardano-api-extended` do it? I think it's the latter, but then, why don't we mention the re-export policy in the paragraph dedicated to the creation of `hydra-cardano-api`?

Reply:
:+1: Moved the re-export policy in the paragraph dedicated to the creation of hydra-cardano-api